atc/db/worker_factory.go

```diff
@@ -57,7 +58,8 @@ var workersQuery = psql.Select(`
 		w.ephemeral
 	`).
 	From("workers w").
-	LeftJoin("teams t ON w.team_id = t.id")
+	LeftJoin("teams t ON w.team_id = t.id").
+	LeftJoin("(SELECT worker_name, COUNT(handle) reported_containers FROM containers GROUP BY worker_name) cs ON w.name = cs.worker_name")
```

Comment:
This should be populating the `expected_containers` count rather than `reported_containers` because the `reported_containers` should be containing the number of containers that the worker heartbeating returns back. (but this is the same as what I mentioned in my previous review comment)

I was thinking here that maybe instead of having the `expected_containers` count be queried out everytime we select for a worker, we could have a method on the `db.Worker` interface called `ExpectedWorker() (int, error)` and it will run a query to find the number of containers for that worker in the containers table. The reason for this is that we can separately run the method to find the reported containers when we need it. If we put the query within the `workersQuery` then everytime we fetch out a `db.Worker` object it will run this query even in places that we don't care about the `expected_containers` (for example, grabbing out all the workers for container placement). The only time we will need the `expected_containers` is in the api so we can just have the api run the method to grab out the number of `expected_containers`. Similar to https://github.com/concourse/concourse/blob/0af59c6a895b1f6acfeaade85d359699ddc29e8b/atc/api/present/worker.go#L21-L24. Let me know what you think!